concepts/observability-configurable-logging/08-revised-api.md

```diff
@@ -0,0 +1,290 @@
+
+### Match expressions should be limited
+
+Support of full match expressions in a `LogPipeline` will harm other pipelines as well as the central `tail` pipeline. People copying code snippets might have match expressions using `*`. It should be prevented.
+-> Match expressions should be defaulted/validated
+
+### Unsupported Plugins
+
+We need a clear distinction between supported/tested setups and unsupported setups. An HTTP plugin usage should be supported with any configuration while a stackdriver plugin usage is unsupported as we have no chance to cover testing.
+-> Introduce a mechanism to detect usage of unsupported plugins
+
+### Dedotting support
+
+There is no filter available for opensearch-typical dedotting. That is a mandatory feature and we planned to solve it via a lua script. On the other hand, the lua filter should be an unsupported feature. So having the convinience of detotting support in the `LogPresetBinding` would result in a lua script in the `LogPipeline`.
+-> Add dedotting support as a feature to the `LogPipeline`
+
+### Meaningfulness of LogPreset
+
+It turned out that an OpenSearch integration via a HTTP plugin required for an internal SAP scenario can be much simpler then expected. A real templating is not needed and probably a plain simple pipeline config will be better suited. Also, what presets to provide, we cannot think of other scenarios which can be delivered. The binding to a secret can be done already in the pipeline itself, workload selectors could be added as well
+-> do we need the LogPreset concept?
+
+### Support of additional inputs
+It turned out that there are typical inputs available which you would like to process as well, mainly the kubelet logs
+-> add a source selector
+
+## Revised API
+
+More and more oppinion/abstraction was added to the `LogPipeline` concept and the meaningfullnes of the `LogPreset` is in the room. We could now try to push down the `LogPipeline` to be more native and accepting that the user can do really stupid things in combination of supported plugins and revise the LogPreset concept.
+Or we push up even more the `LogPipeline` API by introducing the convinience of the LogPreset while keeping native support as a customization option clearly indicating an unsupported scenario.
+This proposal favors the second idea by dropping the LogPreset fully and having only one layer which is already an abstraction. Giving no chance to configure fluent-bit natively. For that you can simply go with a custom installation.
+
+```YAML
+kind: LogPipeline
+apiVersion: telemetry.kyma-project.io/v1alpha1
+metadata:
+  name: OpenSearchHTTP-App
+spec:
+  input:
+    application: # maps to central tail pipeline
+      namespaces: [] # maps to rewrite_tag rule
+      excludeNamespaces: []
+      pods: []
+      excludePods: []
+      containers: []
+      excludeContainers: []
+      matchLabels:
+        app: icke
+    system: {} #maps to systemd based input like kubelet logs
```

Comment:
I have not dealt with that logs at all yet, can you make a proposal? Anyway we will not implement it as a first thing, still it will be good to understand the potential API already

Reply:
The documentation of the systemd input gives the example to use the Systemd_Filter property to select a specific unit. The use-case that I have seen before is to limit the input the the kubelet unit. Giving a list of units is the only use-case that I see at the moment. But I'm fine to postpone the decision on details.